src/apps/integrations/service/integrations.py

```diff
@@ -45,30 +52,79 @@ async def disable_integration(self, query: QueryParams):
             workspace.integrations = None
         await UserWorkspaceCRUD(self.session).save(workspace)
 
-    async def create_integration(
-        self,
-        type: str,
-        params: IntegrationsCreate,
-    ) -> LorisProjects:
-        match type:
+    async def create_integration(self, newIntegration: Integration) -> Integration:
+        match newIntegration.integration_type:
             case AvailableIntegrations.LORIS:
-                return await self._create_integration(type, params)
+                loris_config = cast(LorisIntegration, newIntegration.configuration)
+                expected_keys = ["hostname", "username", "project"]
+                config_dict = loris_config.dict()
+                if None in [config_dict.get(k, None) for k in expected_keys]:
+                    raise UnexpectedPropertiesForIntegration(
+                        provided_keys=list(config_dict.keys()),
+                        expected_keys=expected_keys,
+                        type=AvailableIntegrations.LORIS,
+                    )
+                loris_integration = await LorisIntegrationService(
+                    newIntegration.applet_id, self.session, self.user
+                ).create_loris_integration(
+                    hostname=loris_config.hostname,
+                    username=loris_config.username,
+                    project=loris_config.project,
+                )
+                return Integration(
+                    integration_type=AvailableIntegrations.LORIS,
+                    applet_id=newIntegration.applet_id,
+                    configuration=loris_integration,
+                )
+            case AvailableIntegrations.FUTURE:
+                future_conf = cast(FutureIntegration, newIntegration.configuration)
+                expected_keys = ["endpoint"]
+                config_dict = future_conf.dict()
+                if None in [config_dict.get(k, None) for k in expected_keys]:
+                    raise UnexpectedPropertiesForIntegration(
+                        provided_keys=list(config_dict.keys()),
+                        expected_keys=expected_keys,
+                        type=AvailableIntegrations.LORIS,
```

Comment:
should be `AvailableIntegrations.FUTURE`

Reply:
+1